gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -1,188 +1,97 @@
-local function PlayerSprint(trySprinting, moveKey)
-	if SERVER then return end
-
-	local client = LocalPlayer()
-
-	if trySprinting and not GetGlobalBool("ttt2_sprint_enabled", true) then return end
-	if not trySprinting and not client.isSprinting or trySprinting and client.isSprinting then return end
-	if client.isSprinting and (client.moveKey and not moveKey or not client.moveKey and moveKey) then return end
-
-	client.oldSprintProgress = client.sprintProgress
-	client.sprintMultiplier = trySprinting and (1 + GetGlobalFloat("ttt2_sprint_max", 0)) or nil
-	client.isSprinting = trySprinting
-	client.moveKey = moveKey
+---
+---@class SPRINT
+SPRINT = {
+	-- Set up ConVars
+	convars = {
+		-- @realm shared
+		enabled = CreateConVar("ttt2_sprint_enabled", "1", { FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_REPLICATED }, "Toggle Sprint (Def: 1)"),
+		-- @realm shared
+		multiplier = CreateConVar("ttt2_sprint_max", "0.5", { FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_REPLICATED }, "The speed modifier the player will receive. Will be added on top of 1, so 0.5 => 1.5 speed. (Def: 0.5)"),
+		-- @realm shared
+		consumption = CreateConVar("ttt2_sprint_stamina_consumption", "0.6", { FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_REPLICATED }, "The speed of the stamina consumption (per second; Def: 0.6)"),
+		-- @realm shared
+		regeneration = CreateConVar("ttt2_sprint_stamina_regeneration", "0.3", { FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_REPLICATED }, "The regeneration time of the stamina (per second; Def: 0.3)"),
+		-- @realm shared
+		showCrosshair = CreateConVar("ttt2_sprint_crosshair", "0", { FCVAR_ARCHIVE, FCVAR_NOTIFY, FCVAR_REPLICATED }, "Should the Crosshair be visible while sprinting? (Def: 0)"),
+	},
+}
 
-	net.Start("TTT2SprintToggle")
-	net.WriteBool(trySprinting)
-	net.SendToServer()
+---
+-- Checks if the player is pressing any movement keys and the sprint key at the same time.
+-- @param Player ply
+-- @return boolean
+-- @realm shared
+function SPRINT:PlayerWantsToSprint(ply)
+	local isSprinting = ply:KeyDown(IN_SPEED)
+	local inMovement = ply:KeyDown(IN_FORWARD)
+		or ply:KeyDown(IN_BACK)
+		or ply:KeyDown(IN_MOVERIGHT)
+		or ply:KeyDown(IN_MOVELEFT)
```

Comment:
I know we did it like this before, but do we really want the player to be able to sprint in all directions? Because of your rework I paid attention how other games handle this. And as far as I can tell, they only let you sprint forwards. Which kinda makes sense anyway

Reply:
Mhh dunno we can change it i have no strong opinion on this

Reply:
I think is is weird that you can sprint sideways because that doesn't work in reality.

In MC, if you hold down the sprint key while moving in any direction but forward, the sprinting is just paused